pybamm/models/base_model.py

```diff
@@ -421,30 +421,31 @@ def input_parameters(self):
             self._input_parameters = self._find_symbols(pybamm.InputParameter)
         return self._input_parameters
 
-    def print_parameter_info(self):
-        self._parameter_info = ""
+    def get_parameter_info(self):
+        parameter_info = []
         parameters = self._find_symbols(pybamm.Parameter)
         for param in parameters:
-            self._parameter_info += f"{param.name} (Parameter)\n"
+            parameter_info.append((param.name, "Parameter"))
+
         input_parameters = self._find_symbols(pybamm.InputParameter)
         for input_param in input_parameters:
-            if input_param.domain == []:
-                self._parameter_info += f"{input_param.name} (InputParameter)\n"
+            if not input_param.domain:
+                parameter_info.append((input_param.name, "InputParameter"))
             else:
-                self._parameter_info += (
-                    f"{input_param.name} (InputParameter in {input_param.domain})\n"
-                )
+                parameter_info.append((input_param.name, f"InputParameter in {input_param.domain}"))
+
         function_parameters = self._find_symbols(pybamm.FunctionParameter)
         for func_param in function_parameters:
-            # don't double count function parameters
-            if func_param.name not in self._parameter_info:
-                input_names = "'" + "', '".join(func_param.input_names) + "'"
-                self._parameter_info += (
-                    f"{func_param.name} (FunctionParameter "
-                    f"with input(s) {input_names})\n"
-                )
+            if func_param.name not in [name for name, _ in parameter_info]:
+                input_names =  "', '".join(func_param.input_names)
+                parameter_info.append((func_param.name, f"FunctionParameter with inputs(s) '{input_names}'"))
+
+        return parameter_info
 
-        print(self._parameter_info)
+    def print_parameter_info(self):
```

Comment:
Since the intention of `print_parameter_info` is now to display the information in a more readable way, I would suggest making use of [string formatting](https://stackoverflow.com/a/9536084/14001839) to showcase a pretty-printed table, perhaps something like this:

```
+======================================================+====================+====================================================================================================+
 Parameter [dimensions]                                | Type of parameter  | Parameter inputs                                                                                   
+======================================================+====================+====================================================================================================+
 Initial concentration in electrolyte [mol.m-3]        | Parameter          |
 Lower voltage cut-off [V]                             | Parameter          |
 Separator thickness [m]                               | Parameter          |
 Negative electrode Bruggeman coefficient (electrode)  | Parameter          |
 Upper voltage cut-off [V]                             | Parameter          |
 Positive electrode thickness [m]                      | Parameter          |
 Positive electrode OCP entropic change [V.K-1]        | FunctionParameter  | 'Positive particle stoichiometry', 'Maximum positive particle surface concentration [mol.m-3]')
 Cation transference number                            | FunctionParameter  | 'Electrolyte concentration [mol.m-3]', 'Temperature [K]')
 Electrolyte conductivity [S.m-1]                      | FunctionParameter  | 'Electrolyte concentration [mol.m-3]', 'Temperature [K]')
 ...
+======================================================+====================+====================================================================================================+
```

This is just an initial thought I had, of course there should be a nicer and more maintainable way – since many parameter names are incredibly long, too long to be put in-line in a table cell especially without word wrapping. An easier way would be to print just the two columns (name and type) by keeping the parameter type and inputs combined like they are currently.